hardware/DomoticzHardware.cpp

```diff
@@ -792,7 +792,8 @@ void CDomoticzHardwareBase::SendPercentageSensor(const int NodeID, const uint8_t
 	_tGeneralDevice gDevice;
 	gDevice.subtype = sTypePercentage;
 	gDevice.id = ChildID;
-	gDevice.intval1 = NodeID;
+	gDevice.battery_level = BatteryLevel;
+	gDevice.intval1 = (NodeID << 8) | ChildID;
```

Comment:
This will cause a lot of new Node ID's for various hardware. Please undo this change

Reply:
Battery Percentage is passed along with sDecodeRXMessage, not needed here

Reply:
> 
> 
> This will cause a lot of new Node ID's for various hardware. Please undo this change

True, will revert.

The behavior of the generated NodeID for the Percentage Sensor is different from a lot of other sensors because the ChildID is not part of the generated ID. Could be considered a 'bug'?

Reply:
It is not a 'bug' but it is how it was designed to work. There are more sensors. I guess at that time i thought 255 percentage sensors should be enough per hardware ;)